### DISCUSSION TITLE
ETA for mailmap

### DISCUSSION THREAD

**Discussion by o2sh** [GitoxideLabs/gitoxide#365] [General]
Hi @Byron 
After experiencing some performance issues using `libgit2` in [onefetch](https://github.com/o2sh/onefetch) on large repositories, we thought about trying out  `gitoxide` instead. However, the potential boost in performance may come at a price of losing some features like `mailmap`.

Do you have an idea of when support for mailmap will be added ?

And thanks for the amazing work you've done here 👍

**Comment by Byron** [comment]
Thanks a lot for your interest!

It's a lucky co-incidence the 'oxidization' workstream soon needs to be filled with new tasks, and getting one-fetch to be able to use `gitoxide` more seems like a great way to continue.

Let me first refer to #364 as the tracking ticket for what it would take to fully oxidize git usage.

That said, it looks like getting the git history is the most time-consuming part of the `onefetch` invocation right now. Given that the underlying `git log --pretty=$"format:%cr\t%an" > /dev/null` reports '7.49s user 0.32s system 98% cpu 7.946 total', I tried `ein t h` which also traverses the entire commit graph and does some calculations on it. It results in ' 8.48s user 0.98s system 125% cpu 7.518 total' so I would expect only minor speedups when switching to `gitoxide` there, traversing the graph is inherently single-threaded unfortunately.

By the way, I noticed that `onefetch 2.9` finished the linux kernel (@v5.16) in 15s, whereas `06dbe1a32b3754fb8fbe679c756927e542fba47d` needed ~19.5s.

Looking into this more, I noticed that `git log` isn't used anymore, instead it's a `git2` iteration over the graph which currently takes ~14s give or take. This should go down to ~7s when using `gitoxide` provided that commits aren't looked up twice (this is easy to do by setting an object cache on the ODB handle).

Lastly, as for `mailmap`, I am happy to give it a shot right away, see #366 :).
